Investigate flaky process-mode retry/DLQ recovery failures

## Summary
Process-mode retry/DLQ recovery appears flaky across dependency/perf PR validation. The same failure class has shown up in multiple contexts and should be investigated separately from unrelated feature work.

## Observed symptom
The broker-backed retry/DLQ path intermittently fails to commit the final safe offset after terminal DLQ handling in process mode. In some failing runs the process worker also dies unexpectedly before shutdown/drain finishes.

## Recent evidence
- PR #111 (`perf(control-plane): reduce ordered throughput fixed costs`)
  - all review comments addressed
  - remaining blocker was `e2e (recovery-retry-dlq)` flaking/failing
- PR #105 (`chore(deps): bump pydantic-settings from 2.12.0 to 2.14.0`)
  - rerun still failed on the same broad recovery/DLQ axis

Representative failure signals seen in logs:
- `ProcessWorker[0] died (exitcode=-11). Restarting worker.`
- `Graceful shutdown drain timed out after 5.000s`
- `AssertionError: dlq scenario never committed the final safe offset`

## Likely scope
- process-mode completion timing
- DLQ terminal failure path
- graceful shutdown drain / final safe offset commit
- possible race between retry exhaustion, DLQ publish completion, and commit debounce/flush

## Suggested investigation steps
1. Reproduce with repeated runs of the broker-backed recovery retry/DLQ e2e suite in process mode only.
2. Compare success vs failure logs around:
   - worker exit/crash
   - pending DLQ events
   - dirty commit partition tracking
   - graceful drain timeout
   - final committed offset progression
3. Determine whether the root cause is:
   - flaky test timing only
   - process worker stability issue
   - control-plane commit/DLQ ordering bug
4. Add deterministic regression coverage once the failure mode is isolated.

## Related PRs
- #111
- #105

## Priority
Deferred for now; not the current top priority, but should be revisited before relying on the affected e2e path for merge confidence.

Investigation update after the recent PR #121/develop workflow failure.

## What was observed
The latest failing `e2e (recovery-retry-dlq)` run after PR #121 reproduced the same failure class tracked here:

- run: https://github.com/tomorrow9913/Pyrallel-Consumer/actions/runs/25039640342
- failing job: `e2e (recovery-retry-dlq)`
- failure signals:
  - `Fatal Python error: Segmentation fault`
  - stack in `multiprocessing/queues.py::_feed` and `multiprocessing/reduction.py::dumps`
  - `ProcessWorker[0] died (exitcode=-11). Restarting worker.`
  - `Graceful shutdown drain timed out after 5.000s`
  - `AssertionError: dlq scenario never committed the final safe offset`
  - Python 3.12 warning: forking from a multi-threaded process may lead to deadlocks

## Current conclusion
The evidence points more strongly to CI/test-process contamination around Python 3.12 + multiprocessing/fork/Queue feeder threads than to a deterministic DLQ commit-ordering bug in the current control-plane code.

Cross-checks:
- Current control-plane completion/DLQ unit coverage passed locally.
- Process engine unit coverage passed locally.
- The broker-backed retry/DLQ suite passed locally both as one suite and as isolated exact parametrized cases.
- After isolating the retry/DLQ e2e cases into separate pytest invocations, develop e2e passed.

## Mitigation applied
Committed directly to `develop`:

- `f371f0849fa43fb371af54aa282c4a6730b558f7` — `ci(e2e): isolate recovery retry dlq cases`

This keeps the same workflow check name but runs the retry/DLQ parametrized cases in separate pytest processes and uploads all matching JUnit files.

Verification on develop:
- e2e run: https://github.com/tomorrow9913/Pyrallel-Consumer/actions/runs/25043203259
- `e2e (ordering-kafka-backed)`: success
- `e2e (recovery-rebalance-restart)`: success
- `e2e (recovery-retry-dlq)`: success

## Remaining risk
This is a CI isolation mitigation, not proof that the underlying segfault/fork/Queue feeder risk is fully eliminated. The remaining root-cause work should be tracked separately from merge-blocking feature PRs.

Recommended follow-up scope:
1. Add better process-mode e2e diagnostics around worker death, pending completions, DLQ publish/commit progression, and shutdown drain state.
2. Evaluate whether the process e2e harness should avoid `fork` from a multi-threaded pytest process, or isolate process-mode cases by default.
3. Keep the current isolated workflow layout as the CI baseline because it restored merge confidence for the affected e2e path.

I would keep this issue open if we want to investigate the underlying `exitcode=-11` root cause further; otherwise it can be closed as mitigated by `f371f08`.

--- a/.github/workflows/e2e.yml
+++ b/.github/workflows/e2e.yml
@@ -90,6 +90,8 @@ jobs:
           PYRALLEL_E2E_REQUIRE_BROKER: "1"
         run: |
           mkdir -p .artifacts
+          # Isolate Issue #112 retry/DLQ process-mode cases into fresh pytest
+          # interpreters to avoid fork-after-thread Queue feeder contamination.
           if [ "${{ matrix.suite.name }}" = "recovery-retry-dlq" ]; then
             uv run pytest 'tests/e2e/test_process_recovery.py::test_process_retry_path_commits_only_after_success[async]' -q --maxfail=1 -ra --junitxml=.artifacts/e2e-junit-retry-async.xml
             uv run pytest 'tests/e2e/test_process_recovery.py::test_process_retry_path_commits_only_after_success[process]' -q --maxfail=1 -ra --junitxml=.artifacts/e2e-junit-retry-process.xml

